geopandas/io/tests/test_infer_schema.py

```diff
@@ -0,0 +1,227 @@
+from collections import OrderedDict
+from unittest import TestCase
+
+from hamcrest import has_entries, contains_inanyorder
+from hamcrest.core import assert_that
+from shapely.geometry import Point, Polygon, MultiPolygon, MultiPoint, LineString, MultiLineString
+
+from geopandas import GeoDataFrame
+from geopandas.io.file import infer_schema
+
+
+class TestInferSchema(TestCase):
+
+    # Credit: Polygons below come from Montreal city Open Data portal
+    # http://donnees.ville.montreal.qc.ca/dataset/unites-evaluation-fonciere
+    city_hall_boundaries = Polygon((
+        (-73.5541107525234, 45.5091983609661),
+        (-73.5546126200639, 45.5086813829106),
+        (-73.5540185061397, 45.5084409343852),
+        (-73.5539986525799, 45.5084323044531),
+        (-73.5535801792994, 45.5089539203786),
+        (-73.5541107525234, 45.5091983609661)
+    ))
+    vauquelin_place = Polygon((
+        (-73.5542465586147, 45.5081555487952),
+        (-73.5540185061397, 45.5084409343852),
+        (-73.5546126200639, 45.5086813829106),
+        (-73.5548825850032, 45.5084033554357),
+        (-73.5542465586147, 45.5081555487952)
+    ))
+
+    city_hall_walls = [
+        LineString((
+            (-73.5541107525234, 45.5091983609661),
+            (-73.5546126200639, 45.5086813829106),
+            (-73.5540185061397, 45.5084409343852)
+        )),
+        LineString((
+            (-73.5539986525799, 45.5084323044531),
+            (-73.5535801792994, 45.5089539203786),
+            (-73.5541107525234, 45.5091983609661)
+        ))
+    ]
+
+    city_hall_entrance = Point(-73.553785, 45.508722)
+    city_hall_balcony = Point(-73.554138, 45.509080)
+    city_hall_council_chamber = Point(-73.554246, 45.508931)
+
+    point_3D = Point(-73.553785, 45.508722, 300)
+    linestring_3D = LineString((
+        (-73.5541107525234, 45.5091983609661, 300),
+        (-73.5546126200639, 45.5086813829106, 300),
+        (-73.5540185061397, 45.5084409343852, 300)
+    ))
+    polygon_3D = Polygon((
+        (-73.5541107525234, 45.5091983609661, 300),
+        (-73.5535801792994, 45.5089539203786, 300),
+        (-73.5541107525234, 45.5091983609661, 300)
+    ))
+
+    def test_infer_schema_when_dataframe_has_only_points(self):
+        df = GeoDataFrame(
+            {},
+            crs={'init': 'epsg:4326', 'no_defs': True},
+            geometry=[self.city_hall_entrance, self.city_hall_balcony]
+        )
```

Comment:
Small note here: this can be simplified a bit, as the `{}` is not needed (if you only specify `geometry`, you get a dataframe with only a 'geometry' column). And for those tests, I think the `crs` is not necessarily needed?

So something like `df = GeoDataFrame(geometry=[self.city_hall_entrance, self.city_hall_balcony])` should be sufficient.